makeflow/src/dag_node.c

```diff
@@ -49,22 +50,213 @@ struct dag_node *dag_node_create(struct dag *d, int linenum)
 	return n;
 }
 
-const char *dag_node_state_name(dag_node_state_t state)
+struct dag_node_footprint *dag_node_footprint_create()
 {
-	switch (state) {
-	case DAG_NODE_STATE_WAITING:
-		return "waiting";
-	case DAG_NODE_STATE_RUNNING:
-		return "running";
-	case DAG_NODE_STATE_COMPLETE:
-		return "complete";
-	case DAG_NODE_STATE_FAILED:
-		return "failed";
-	case DAG_NODE_STATE_ABORTED:
-		return "aborted";
-	default:
-		return "unknown";
+	struct dag_node_footprint *f;
+
+	f = malloc(sizeof(struct dag_node_footprint));
+	memset(f, 0, sizeof(struct dag_node_footprint));
```

Comment:
Just a little thing:

You can either do: f = calloc(1, sizeof(struct dag_node_footprint)); which automatically zeros f, or
use bzero(f, sizeof(struct dag_node_footprint)); instead of memset(..., 0, ...